tests/test_typing.py

```diff
@@ -101,6 +105,282 @@ class UnknownType:
             str(cm.exception)
         )
 
+    def test_handle(self):
```

Comment:
Could you move the tests to a new test case class, please?

Reply:
will do!